stripecardscan/src/main/java/com/stripe/android/stripecardscan/scanui/CameraView.kt

```diff
@@ -0,0 +1,219 @@
+ *  [R.styleable.CameraView_viewFinderType] - a enum to decide the type of [ViewFinderType], which dictates the aspect ratio of [viewFinderWindowView].
+ *  [R.styleable.CameraView_borderDrawable] - an optional reference for [viewFinderBorderView]'s drawable resource.
+ *
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class CameraView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+    /**
+     * a [FrameLayout] to display the preview frame of camera feed, has the same size of the [CameraView].
+     */
+    val previewFrame: FrameLayout by lazy { FrameLayout(context) }
+
+    /**
+     * an optional [ViewFinderBackground] to draw background and reveals a center viewFinder on previewFrame, has the same size of the [CameraView].
+     */
+    val viewFinderBackgroundView: ViewFinderBackground by lazy {
+        ViewFinderBackground(context)
+    }
+
+    /**
+     * a [View] to highlight a sub area from previewFrame, the [Rect] of this view is used to crop the highlighted area from the preview.
+     */
+    val viewFinderWindowView: View by lazy { View(context) }
+
+    /**
+     * an optional [ImageView] to draw the border of [viewFinderWindowView], exposed for applying animations.
+     */
+    val viewFinderBorderView: ImageView by lazy { ImageView(context) }
+
+    @DrawableRes
+    private var borderDrawable: Int = NO_BORDER
+
+    private var isCreatedFromXML: Boolean = false
+
+    private var viewFinderType: ViewFinderType = ViewFinderType.CreditCard
+
+    /**
+     * The type of viewfinder, decides if [viewFinderBackgroundView] should be drawn and the aspect ratio of [viewFinderWindowView]
+     */
+    private enum class ViewFinderType(
+        val aspectRatio: String
+    ) {
+        /**
+         * [Fill] doesn't draw [viewFinderBackgroundView] and fills [viewFinderWindowView] with parent
+         */
+        Fill(FILL_ASPECT_RATIO),
+
+        /**
+         * [CreditCard] draws [viewFinderBackgroundView] and draws [viewFinderWindowView] in center with aspect ratio [CREDIT_CARD_ASPECT_RATIO]
+         */
+        CreditCard(CREDIT_CARD_ASPECT_RATIO),
+
+        /**
+         * [ID] draws [viewFinderBackgroundView] and draws [viewFinderWindowView] in center with aspect ratio [ID]
+         */
+        ID(ID_ASPECT_RATIO),
+
+        /**
+         * [Passport] draws [viewFinderBackgroundView] and draws [viewFinderWindowView] in center with aspect ratio [Passport]
+         */
+        Passport(PASSPORT_ASPECT_RATIO)
+    }
+
+    init {
+        context.withStyledAttributes(
+            attrs,
+            R.styleable.CameraView
+        ) {
+            // if it's null then the view is created programmatically by Bouncer, default to credit
+            // card specifics
+            // TODO(awush): migrate Bouncer to use xml with type [ViewFinderType.CreditCard]
```

Comment:
@awush-stripe fyi added two TODOs here for you for migrating to XML, after which we could remove the special case for programmatically creating :)